File: src/main/java/game/Cavalo.java
package game;

public class Cavalo extends Peca {
    public Cavalo(String cor) {
        super(cor);
        this.nomePeca = "Cavalo";
    }

    @Override
    public boolean isMovimentoValido(Tabuleiro tabuleiro, int linhaInicial, int colunaInicial, int linhaFinal, int colunaFinal) {
        Peca pecaDestino = tabuleiro.getPeca(linhaFinal, colunaFinal);

        if (pecaDestino != null && pecaDestino.getCor().equals(this.getCor())) {
            return false;
        }

        int diffLinha = Math.abs(linhaFinal - linhaInicial);
        int diffColuna = Math.abs(colunaFinal - colunaInicial);

        boolean ehMovimentoL = (diffLinha == 2 && diffColuna == 1) || (diffLinha == 1 && diffColuna == 2);

        // Como o cavalo pula, não precisamos checar o caminho.
        return ehMovimentoL;
    }

    @Override
    public String toString() {
        return "C{}";
    }
}

File: Projeto JS/src/game/Cavalo.js
import { Peca } from './Peca.js';

export class Cavalo extends Peca {
    constructor(cor) {

        super(cor); 
        this.nomePeca = "Cavalo";
    }

   
    isMovimentoValido(tabuleiro, linhaInicial, colunaInicial, linhaFinal, colunaFinal) {
       
        const pecaDestino = tabuleiro.getPeca(linhaFinal, colunaFinal);

        if (pecaDestino !== null && pecaDestino.getCor() === this.getCor()) {
            return false;
        }

        const diffLinha = Math.abs(linhaFinal - linhaInicial);
        const diffColuna = Math.abs(colunaFinal - colunaInicial);

        const ehMovimentoL = (diffLinha === 2 && diffColuna === 1) || (diffLinha === 1 && diffColuna === 2);

        return ehMovimentoL;
    }

    toString() {
        return "C{}";
    }
}